packages/next/src/build/analysis/get-page-static-info.ts

```diff
@@ -444,29 +447,30 @@ function warnAboutExperimentalEdge(apiRoute: string | null) {
   apiRouteWarnings.set(apiRoute, 1)
 }
 
-const warnedUnsupportedValueMap = new LRUCache<string, boolean>({ max: 250 })
+export let hadUnsupportedValue = false
 
 function warnAboutUnsupportedValue(
   pageFilePath: string,
   page: string | undefined,
   error: UnsupportedValueError
 ) {
-  if (warnedUnsupportedValueMap.has(pageFilePath)) {
-    return
-  }
+  hadUnsupportedValue = true
 
-  Log.warn(
+  const message =
     `Next.js can't recognize the exported \`config\` field in ` +
-      (page ? `route "${page}"` : `"${pageFilePath}"`) +
-      ':\n' +
-      error.message +
-      (error.path ? ` at "${error.path}"` : '') +
-      '.\n' +
-      'The default config will be used instead.\n' +
-      'Read More - https://nextjs.org/docs/messages/invalid-page-config'
-  )
-
-  warnedUnsupportedValueMap.set(pageFilePath, true)
+    (page ? `route "${page}"` : `"${pageFilePath}"`) +
+    ':\n' +
+    error.message +
+    (error.path ? ` at "${error.path}"` : '') +
+    '.\n' +
+    'Read More - https://nextjs.org/docs/messages/invalid-page-config'
+
+  // for a build wait to log all errors before exiting
+  if (process.env.NEXT_PHASE !== PHASE_PRODUCTION_BUILD) {
+    Log.error(message)
```

Comment:
I don't know how `Log.error` works. I was expecting it to log the error and at the end of the phase, we error if we called any `Log.error`. If that's the case, shouldn't this be 
```suggestion
  // for a build wait to log all errors before exiting
  if (process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD) {
    Log.error(message)
```

Also: Do we have a different phase for development builds or is `PHASE_PRODUCTION_BUILD` just a remnant from when we didn't have `experimental.allowDevelopmentBuild `?

Reply:
The `Log.error` is just different prefix from `Log.warn`, we only have `PHASE_PRODUCTION_BUILD` which is set in build here https://github.com/vercel/next.js/blob/e740c48a1f68df6a2cf3696a4170d4dc1a9198aa/packages/next/src/build/index.ts#L1814

Reply:
I don't understand the comment then. We are now throwing in build. Does throwing means we collect all errors before exiting and Log.error would mean we don't wait?

Reply:
`Log.error` is just `console.error` which we want in a build and then after we call `getStaticPageInfo` for all paths if any had invalid values we actually exit here https://github.com/vercel/next.js/blob/51892ebbfd1c31f36edd6d09b1385a52c84b0aa3/packages/next/src/build/index.ts#L2309-L2314

In dev we just throw since HMR handles it

Reply:
> Log.error is just console.error which we want in a build

Sorry, I mistyped. The code says: outside of build, log, otherwise, throw. It's the equivalent to 

```
// for a build wait to log all errors before exiting
if (process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD) {
  throw new Error(message)
} else {
  console.error(message)
}
```
The comment above does not match the code.

Reply:
This "phase" concept is unclear to me. I logged the phase during the build in the mentioned test and get this output:
```
Checking validity of types ...
       Creating an optimized production build ...
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
    { phase: undefined }
     ✓ Compiled successfully
       Collecting page data ...
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
    { phase: 'phase-production-build' }
```

the phase is `undefined` during "creating a production build" and `phase-production-build` during "collecting page data". I would expect the phase to be `phase-production-build` during "creating a production build" and `phase-collecting-page-data` during "collecting page data". Or I would expect it to be called something different like "entrypoint" or something. A "phase" is like a "stage" i.e. something that is part of a larger process. But the comment indicates `next build` has a single phase

Reply:
It's also logging the error multiple times. Not sure if new or old behavior:
```
    ⨯ Next.js can't recognize the exported `config` field in route "/array-spread-operator":
    Unsupported spread operator in the Array Expression at "config.runtime".
    Read More - https://nextjs.org/docs/messages/invalid-page-config
     ⨯ Next.js can't recognize the exported `config` field in route "/array-spread-operator":
    Unsupported spread operator in the Array Expression at "config.runtime".
    Read More - https://nextjs.org/docs/messages/invalid-page-config
     ⨯ Next.js can't recognize the exported `config` field in route "/array-spread-operator":
    Unsupported spread operator in the Array Expression at "config.runtime".
    Read More - https://nextjs.org/docs/messages/invalid-page-config
```

Reply:
Follow-ups in https://github.com/vercel/next.js/pull/68696

Reply:
Good catch!